### PR TITLE
Migrate to SpringDoc.

### PR DESCRIPTION
**Pull request by mshima**
Depends on https://github.com/jhipster/generator-jhipster/pull/17194

### CODE REVIEW

File: jhipster-framework/pom.xml [hunk 1]
```diff
@@ -48,18 +48,18 @@
...
-            <artifactId>springfox-oas</artifactId>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-webmvc-core</artifactId>
             <optional>true</optional>
         </dependency>
         <dependency>
-            <groupId>io.springfox</groupId>
-            <artifactId>springfox-swagger2</artifactId>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-webflux-core</artifactId>
```
Comment (mraible) [inline jhipster-framework/pom.xml:57]: Using this dependency, I get an error when running `mvn verify` on the `webflux-mongodb` example:

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springdoc.core.GenericResponseService]: Factory method 'responseBuilder' threw exception; nested exception is java.lang.NoClassDefFoundError: Could not initialize class org.springdoc.core.GenericResponseService
```

If I change it to `springdoc-openapi-webflux-ui`, all tests pass. This seems to match what the [documentation recommends](https://springdoc.org/#spring-webflux-support). Therefore, I'm going to update this dependency.

Comment (mshima) [reply jhipster-framework/pom.xml:57]: `springdoc-openapi-webflux-ui` pulls in swagger-ui webjar, we already have swagger ui at the client side.

Comment (mraible) [reply jhipster-framework/pom.xml:57]: Yeah, `webflux-ui` causes startup issues (even though `mvn verify passes). Strangely, if I use what we have now, I get the following error on startup.

```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 
'openApiResource' defined in class path resource 
[org/springdoc/webflux/core/SpringDocWebFluxConfiguration.class]: Unsatisfied dependency expressed through 
method 'openApiResource' parameter 2; nested exception is 
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'responseBuilder' 
defined in class path resource [org/springdoc/webflux/core/SpringDocWebFluxConfiguration.class]: Bean 
instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: 
Failed to instantiate [org.springdoc.core.GenericResponseService]: Factory method 'responseBuilder' threw 
exception; nested exception is java.lang.NoClassDefFoundError: javax/servlet/ServletException
```

Comment (mshima) [reply jhipster-framework/pom.xml:57]: Should be fixed now.

Comment (mshima) [reply jhipster-framework/pom.xml:57]: I spoke too soon too. 😆.
The problem is springdoc-openapi-hateoas now.
Wondering if data-rest is non-reactive only.

Comment (mshima) [reply jhipster-framework/pom.xml:57]: fixed

**Comment by mraible** [comment]
Closing and reopening to run CI again.

**Comment by mshima** [comment]
Tests are the same from the generator-jhipster side. Should consider them only.

**Comment by mraible** [comment]
Can we merge this then? I'm eager to migrate to Spring Boot 2.6 before I
start working on
https://github.com/jhipster/generator-jhipster/issues/16498.

**Comment by mshima** [comment]
I think it’s ready to me merged.

**Pull request merged**